armi/tests/mockRunLogs.py

```diff
@@ -44,10 +47,55 @@ def __enter__(self):
 
     def __exit__(self, exception_type, exception_value, traceback):
         runLog.LOG = self.originalLog
-        runLog.LOG.setStreams(runLog.LOG._outputStream, runLog.LOG._errStream)
+
+    def log(self, msgType, msg, single=False, label=None):
+        """
+        Add formatting to a message and handle its singleness, if applicable.
+
+        This is a wrapper around logger.log() that does most of the work and is
+        used by all message passers (e.g. info, warning, etc.).
+        """
+        # the message label is only used to determine unique for single-print warnings
+        if label is None:
+            label = msg
+
+        # Skip writing the message if it is below the set verbosity
+        msgVerbosity = self._logLevels[msgType][0]  # pylint: disable=protected-access
+        if msgVerbosity < self._verbosity:
+            return
+
+        # Skip writing the message if it is single-print warning
+        if single and self._msgHasAlreadyBeenEmitted(label, msgType):
+            return
```

Comment:
Is this handled by the dedup Filter now? Happy with this if there is a good reason to have it for testing purposes, but seems like we'd want to keep this as close to the base impl as possible.

Reply:
Naw.  Since the `MockRunLogs.BufferLog` class doesn't actually _use_ the `logging.Logger` objects, it also doesn't use the `DedupFilter`. It uses plain string streams.  

The old `runLog` system wrote text files. Well, those were annoying to parse for testing, so `BufferLog` existed to skip the hard files and write to a string buffer, which could be used much more easily for testing.

The new `runLog` system writes using the `logging` module, which presents it's own problems because PyTest automatically intercepts all `logging` statements.  So now this class is used to bypass Python's standard library `logging` module, and write directly to a in-memory string stream, so we can test against that.

If I were to change anything, I guess I would want to refactor `_RunLog` to make things like this `BufferLog` class shorter/easier to write.

Reply:
sweet, thanks for the context